### PR TITLE
Generate default kind components when they don't exist

### LINKED ISSUE #56: Generate default kind components
It would allow us to remove a lot of boilerplate if we could generate the `--default` kind components when they are not doing anything special. These components should just be clones of the base `{{ui-kind}}`.

### PR DESCRIPTION
**Pull request by HeroicEric**
Closes #56

### CODE REVIEW

File: addon/styles/components/ui-dropbutton-trigger--default.scss [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@component() {
```
Comment (HeroicEric) [inline addon/styles/components/ui-dropbutton-trigger--default.scss:1]: this file currently needs to exist for the generation to happen

File: lib/js-preprocessor/broccoli-generate-components.js [hunk 2]
```diff
@@ -76,15 +76,16 @@ var componentsToGenerate = function(baseDir, scss) {
...
-    var defaultFile = fileName.replace(/(.+)--(.+)\.js$/, '$1--default.js');
-    var defaultFileExists = existsSync(path.join(baseDir, defaultFile));
-
-    if (defaultFileExists) {
-      components.push({
-        component: fileName,
-        default: defaultFile
-      });
-    }
+    var baseKindFile = 'untitled-ui/components/ui-kind';
```
Comment (HeroicEric) [inline lib/js-preprocessor/broccoli-generate-components.js]: Is there a better way to construct this path?

File: lib/js-preprocessor/index.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ var funnel = require('broccoli-funnel');
 module.exports = function(tree, options) {
   var generated = new GenerateComponents(tree, options);
 
-  var merged = mergeTrees([tree, generated]);
+  var merged = mergeTrees([tree, generated], { overwrite: true });
```
Comment (HeroicEric) [inline lib/js-preprocessor/index.js]: I was running into issues when this wasn't set. What are the repercussions of this?

File: lib/js-preprocessor/broccoli-generate-components.js (hunk 2 above)
Comment (ianstarz) [inline lib/js-preprocessor/broccoli-generate-components.js]: I think `this.prefix` might be what you're looking for

```
var baseKindFile = path.join(this.prefix, 'ui-kind');
```

File: lib/js-preprocessor/index.js (hunk 3 above)
Comment (ianstarz) [inline lib/js-preprocessor/index.js]: We should try and not use this. If someone writes a base component manually, we don't want it to be overridden by the auto generated component

Comment (HeroicEric) [inline lib/js-preprocessor/index.js]: This is the error I'm getting without it https://gist.github.com/HeroicEric/f8b25ee502bd3dfa59f5c5d8e18c2fa8

I'm not sure why yet but this function is being called twice. Is that expected?

File: lib/js-preprocessor/broccoli-generate-components.js [hunk 4]
```diff
@@ -60,8 +60,9 @@ GenerateComponents.prototype.appComponentFor = function(fileName) {
   return "export { default } from '" + defaultModule + "';\n";
 };
 
-var componentsToGenerate = function(baseDir, scss) {
+var componentsToGenerate = function(baseDir, scss, modulePrefix) {
   var scssFiles = walkSync(baseDir, { globs: scss });
+  var baseKindFile = path.join(modulePrefix, 'components', 'ui-kind').replace('modules/', '');
```
Comment (HeroicEric) [inline lib/js-preprocessor/broccoli-generate-components.js:65]: @ianstarz i had to do this to get the import paths to be correct in the addon component files :(

Comment (ianstarz) [inline lib/js-preprocessor/broccoli-generate-components.js:65]: ah my mistake, you can actually pass in `this.moduleName` and use that as the prefix instead. I passed in both to the broccoli plugin for this reason. This is so we can config this from the index.js file (broccoli shouldn't have knowledge of what ember's namespacing is)

**Pull request merged**